pylint/checkers/classes/class_checker.py

```diff
@@ -467,7 +478,9 @@ def _is_attribute_property(name, klass):
     return False
 
 
-def _has_same_layout_slots(slots, assigned_value):
+def _has_same_layout_slots(
+    slots: list[nodes.Const], assigned_value: nodes.Name
```

Comment:
Is this correct? Some of the patterns in the calling function seem to indicate this can also be `list[None]` for example.

Reply:
I think you're right, it would explain the ``if all(
            first_slot and second_slot and first_slot.value == second_slot.value
            for (first_slot, second_slot) in zip_longest(slots, other_slots)
        ):`` later